engine/classes/Elgg/PersistentLoginService.php

```diff
@@ -158,10 +160,15 @@ public function getUserFromHash($hash) {
 			return null;
 		}
 
-		$hash = $this->db->sanitizeString($hash);
-		$query = "SELECT guid FROM {$this->table} WHERE code = '$hash'";
+		
+		$query = "SELECT guid
```

Comment:
Why not use QueryBuilder?
```php
$qb = Select::fromTable($this->table);
$qb->select('guid')
       ->where($qb->compare('code', '=', $hash, 'string'));
```

And you can do same for `Update`, `Delete`, `Insert`. Ultimately, I think it's more readable and less prone to typos. Also, if you ever need to do a unit test, it's easier to mock queries.